source/io.c

```diff
@@ -826,24 +852,18 @@ static struct aws_input_stream *aws_input_stream_new_from_py(PyObject *py_self)
         return NULL;
     }
 
+    /* The lifetime of the input stream will be exactly the same as python object passed in */
     struct aws_allocator *alloc = aws_py_get_allocator();
     struct aws_input_stream_py_impl *impl = aws_mem_calloc(alloc, 1, sizeof(struct aws_input_stream_py_impl));
     if (!impl) {
         return NULL;
     }
 
-    impl->base.allocator = alloc;
+    impl->allocator = alloc;
     impl->base.vtable = &s_aws_input_stream_py_vtable;
-    impl->base.impl = impl;
-    impl->self_proxy = PyWeakref_NewProxy(py_self, NULL);
```

Comment:
SHOULD we still do some kind of weakref here?
Is Python allowed to move the Python object in memory if we don't tell it that we're keeping C references via INCREF or weakref or anything?

(remind me to look this up after)

Reply:
nevermind. what you're doing seems safe. The docs on C-extensions talk about "borrowing" a reference without calling Py_INCREF, seems like you just need to be very careful

https://docs.python.org/3/extending/extending.html#reference-counting-in-python